plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -1694,6 +1694,7 @@ public void pause(final boolean v) throws IOException {
                                 LOGGER.log(Level.WARNING, null, t);
                             }
                         });
+                        cpsExec.getOwner().getListener().getLogger().close();
```

Comment:
Is `getOwner()` guaranteed to be non-null? `getListener()`? In principle SpotBugs should know but sometimes it seems to miss these things.

Reply:
I think `CpsFlowExecution.owner` is guaranteed to be non-null. It's set [here](https://github.com/jenkinsci/workflow-cps-plugin/blob/7b84f995ad630b07b7ef193e31a87dd9db85770f/plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java#L406) and [here](https://github.com/jenkinsci/workflow-cps-plugin/blob/7b84f995ad630b07b7ef193e31a87dd9db85770f/plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java#L721) (called [here](https://github.com/jenkinsci/workflow-job-plugin/blob/e6d865025906842fac129cab52ad00722eaeb124/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java#L718)).

`getListener` may return `WorkflowRun.NULL_LISTENER`, which is harmless to close, or throw an exception, but this code already handles exceptions, and I made this be the final operation in the `try` block so that other more important cleanup still runs even if this throws.